internal/retry/retry.go

```diff
@@ -0,0 +1,122 @@
+//    \\ SPIKE: Secure your secrets with SPIFFE.
+//  \\\\\ Copyright 2024-present SPIKE contributors.
+// \\\\\\\ SPDX-License-Identifier: Apache-2.0
+
+package retry
```

Comment:
Can we move this functionality to `./pkg/retry` instead?

I think it's a feature that can be consumed externally if other people need a similar functionality too.